### PR TITLE
add more tests

### PR DESCRIPTION
**Pull request by omerb01**
- added `map()` with different iterdata types tests
- added "return futures" tests
- added `chunk_n` tests
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by omerb01** [comment]
@JosepSampe some tests dont pass.

**Comment by gilv** [comment]
@omerb01 are all new tests pass or some still fails?

**Comment by omerb01** [comment]
@gilv these tests fail:
- `test_map()`
- `test_multiple_executions()`
- `test_internal_executions()`
- `test_chunks_bucket()`
- `test_chunks_bucket_one_reducer_per_object()`

the rest of them pass

**Comment by gilv** [comment]
@omerb01 why they fail?

File: pywren_ibm_cloud/tests.py [hunk 1]
```diff
@@ -43,315 +46,362 @@
...
+
+        pw = pywren.function_executor(config=CONFIG)
+        generator_iterdata = range(2)
+        pw.map(TestMethods.hello_world, generator_iterdata)
+        result = pw.get_result()
+        self.assertEqual(result, ['Hello World!'] * 2)
+
+        pw = pywren.function_executor(config=CONFIG)
+        listDicts_iterdata = [{'x': 2, 'y': 8}, {'x': 2, 'y': 8}]
+        pw.call_async(TestMethods.simple_map_function, listDicts_iterdata)
```
Comment (JosepSampe) [inline pywren_ibm_cloud/tests.py]: @omerb01 It should be pywren.map(), right?

**Comment by JosepSampe** [comment]
With #232 only fails **test_chunks_bucket** and **test_chunks_bucket_one_reducer_per_object**.

With 1MB minimum chunk size:
```
======================================================================
FAIL: test_chunks_bucket (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/josep/data/dev-workspace/pywren-ibm-cloud/pywren_ibm_cloud/tests.py", line 380, in test_chunks_bucket
    self.assertEqual(len(futures), 8)
AssertionError: 6 != 8

======================================================================
FAIL: test_chunks_bucket_one_reducer_per_object (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/josep/data/dev-workspace/pywren-ibm-cloud/pywren_ibm_cloud/tests.py", line 398, in test_chunks_bucket_one_reducer_per_object
    self.assertEqual(len(futures), 12)
AssertionError: 10 != 12
```
With NO minimum chunk size:
```
======================================================================
FAIL: test_chunks_bucket (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/josep/data/dev-workspace/pywren-ibm-cloud/pywren_ibm_cloud/tests.py", line 380, in test_chunks_bucket
    self.assertEqual(len(futures), 8)
AssertionError: 11 != 8

======================================================================
FAIL: test_chunks_bucket_one_reducer_per_object (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/josep/data/dev-workspace/pywren-ibm-cloud/pywren_ibm_cloud/tests.py", line 398, in test_chunks_bucket_one_reducer_per_object
    self.assertEqual(len(futures), 12)
AssertionError: 15 != 12
```

File: pywren_ibm_cloud/tests.py (hunk 1 above)
Comment (omerb01) [reply pywren_ibm_cloud/tests.py]: @JosepSampe yep, thanks

**Comment by omerb01** [comment]
@JosepSampe can merge?

**Comment by JosepSampe** [comment]
@omerb01 What about test_chunks_bucket and test_chunks_bucket_one_reducer_per_object? what should be the correct number of futures?

**Comment by gilv** [comment]
@omerb01 tests fails...do you know why?

**Comment by JosepSampe** [comment]
@gilv It is due the logs I post above. The number of futures should be 6 and 10

**Comment by gilv** [comment]
@JosepSampe that's i got :) i just try to figure out why it's written 6 and not 10...was there mistake in the tests or it's some issue in the code...

**Comment by omerb01** [comment]
@JosepSampe @gilv 
With 1MB minimum chunk size, it should be 8 and 12 since 2 objects in the dataset are greater than 1MB
without this limit, your results are correct.
what do you suggest? to stay with the 1MB limit or remove that limit?

**Comment by JosepSampe** [comment]
@gilv What do you think? should we keep 1MB limit chink size or not? In my opinion I think it makes no sense let (inexpert) users to create chunks lower than 1MB, since it can become in performance degradation.

**Comment by omerb01** [comment]
@gilv I agree with @JosepSampe

**Comment by gilv** [comment]
@omerb01 @JosepSampe I already agree...no point to create such small chunk sizes

**Comment by omerb01** [comment]
@JosepSampe @gilv regarding #184 I suggest to let the user define the minimum size.
I firstly have suggested adding this feature because in EMBL project I got multiple txt files with very small sizes (~1KB), but each file generates an enormous amount of data. so I would like to get more control over how I split the data.

**Comment by JosepSampe** [comment]
@omerb01 @gilv Ok, that makes sense, so I will delete the limit

**Comment by JosepSampe** [comment]
@omerb01 Change futures to 11 and 15 and it can be already merged

**Comment by JosepSampe** [comment]
@omerb01 Can you change the limit? So I won't have to submit another PR.

Only need to change this [line](https://github.com/pywren/pywren-ibm-cloud/blob/master/pywren_ibm_cloud/job/partitioner.py#L26) to this: `CHUNK_SIZE_MIN = 0*1024  # 0MB`

**Comment by JosepSampe** [comment]
@gilv ready ;)

**Pull request merged**